### PR TITLE
add tolerance to toEndButton

### PR DESCRIPTION
**Pull request by maltee1**
Now with (hopefully reasonable) tolerance.
Do we still need the delay as well?

### CODE REVIEW

**Comment by deepbluev7** [comment]
I think the tolerance is too big, it should be a few pixel, not like a whole page or so. And yeah, I think the delay is still nice to have, although we could maybe lower it a bit.

**Comment by maltee1** [comment]
The rationale was that if you're only one page away from the end, you don't need a button to go there. The button's purpose is to save scrolling, after all. People might find it annoying if it pops up when it isn't needed. I suppose half a page could also work, but a few pixels is too little.

**Comment by deepbluev7** [comment]
Well, I usually use such buttons to know, that I am not at the bottom. But I see, why it is so far up now. Your patch seems to also cause a scroll down button everytime a few people have sent messages though?

**Comment by maltee1** [comment]
apparently I just didn't finish the code...
I understand your reasoning why you want the down button to appear after just a couple of pixels. Feel free to change it.

File: resources/qml/MessageView.qml [hunk 1]
```diff
@@ -49,8 +49,12 @@ Item {
         onCountChanged: {
             // Mark timeline as read
             if (atYEnd && room) model.currentIndex = 0;
-
+            lastItem = itemAtIndex(0)
```
Comment (deepbluev7) [inline resources/qml/MessageView.qml:52]: This delegate might be deleted in the meantime.

**Comment by deepbluev7** [comment]
This seems to have a lot of random issues for me. When changing rooms it sometimes shows this for a few seconds. Sometimes it doesn't go away at all...

**Comment by maltee1** [comment]
meh, I suppose the tolerance isn't really necessary anyway :D
I'm a bit real-life busy these days, so little time to spare for our beloved nheko..

**Comment by deepbluev7** [comment]
I think this adds more annoyance than it fixes, so I'll just close it for now and we can revisit it at a later date. Maybe we can just fix the scroll positioning issue instead.

**Pull request closed without merging**